plugin/internal/remotecall/remotecall.go

```diff
@@ -0,0 +1,227 @@
+package remotecall
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	UserAgentKey        = "User-Agent"
+	UserAgentSSHGateway = "SSH-gateway"
+	AcceptKey           = "Accept"
+	ContentType         = "Content-Type"
+	ApplicationJson     = "application/json"
+
+	AuthTokenUserClusterMapping = "authToken"
+	AuthKeyAuthenticator        = "Authorization"
+	IdentitySSHGateway          = "SSH-gateway"
+	JwtValidity                 = 5 * time.Second
+)
+
+type RemoteCall struct {
+	userClusterNameURL *url.URL
+	userClusterToken   string
+
+	clusterNameInClusterAuthenticatorURL map[string]*url.URL
+	serviceJwtProvider                   map[string]*ServiceJWTProvider
+
+	// keeping it string since these won't have http
+	clusterNameInClusterServiceClusterURL map[string]string
```

Comment:
so many clusters in there it's hard to read and wrap your head around :D 
it's `clusterNameToUpstreamServiceURL`, right?

Reply:
changed a bit